### ISSUE TITLE
Cache branch/PR state

### ISSUE THREAD

**Issue by rarkins** [renovatebot/renovate#10232]
A branch can be in three main states:
- Awaiting Branch creation
- Branch created, but awaiting PR creation
- PR open

Reasons why a branch doesn't exist:
- Schedule
- Pending checks (internal))
- Rate limited

Reasons why a PR doesn't exist:
- Schedule (?)
- Awaiting passing tests (prCreation=status-success)
- Awaiting test completion (prCreation=not-pending)
- Needs Approval (prCreation=approval)
- Rate limited

States which a PR can be in:
- Open
- Awaiting automerge
- Automerged with comment

This can also be used for more accurate Dependency Dashboard

**Comment by rarkins** [comment]
Dashboard mockup idea:

This issue contains a list of Renovate updates and their statuses.

## Awaiting Branch Creation

These branches can be created by Renovate on-demand once you click their checkbox below.

 - [ ] build(deps): update dependency @aws-sdk/client-ecr to v3.16.0 ☑️ 
 - [ ] build(deps): update dependency azure-devops-node-api to v11 ☑️ 
 - [ ] build(deps): update dependency crypto-random-string to v4 ☑️ 
 - [ ] build(deps): update dependency markdown-table to v3 ☑️ 
 - [ ] build(deps): update dependency p-map to v5 ☑️ 
 - [ ] build(deps): update dependency p-queue to v7 ☑️ 
 - [ ] build(deps): update dependency strip-ansi to v7 ☑️ 
 - [ ] chore(deps): update node.js to v16 ☑️ 
 - [ ] docs: update references to renovate/renovate 📆 
 - [ ] docs: update references to renovate/renovate 📆 
 - [ ] chore(deps): lock file maintenance 📆 
 - [ ] chore(deps): update node.js to ed0af90 👶 
 - [ ] build(deps): update dependency node-html-parser to v3.3.4 👶 
 - [ ] chore(deps): update actions/cache action to v2.1.6 👶 
 - [ ] chore(deps): update dependency @types/eslint to v7.2.12 👶 
 - [ ] chore(deps): update dependency @types/node to v14.17.1 👶 
 - [ ] chore(deps): update jest monorepo (`@jest/globals`, `@jest/reporters`, `@jest/test-result`, `jest`, `jest-circus`) 👶 
 - [ ] build(deps): update dependency cacache to v15.2.0 👶 

## Branch Exists But Awaiting PR Creation

Tick any checkbox below to manually force creation of any of the following:

 - [ ] build(deps): update dependency detect-indent to v6.1.0 🚦 
 - [ ] build(deps): update dependency shlex to v2.1.0 🚦 
 - [ ] chore(deps): update dependency jest-junit to v12.1.0 🚦 

## Open

 - [ ] chore(deps): update dependency typescript to v4.3.2 🚀 
 - [ ] chore(deps): update linters (`@typescript-eslint/eslint-plugin`, `@typescript-eslint/parser`, `eslint-plugin-import`)

## Closed / Ignored

 - [ ] chore(deps): update node.js to v14.17.0

Reference:
☑️ Needs manual approval in dashboard
📆 Awaiting scheduled branch creation time
👶 Package is awaiting stabilityDays completion
🚦 Rate limited
🚀 Automerge enabled

**Comment by viceice** [comment]
Maybe move the emojis after checkbox for better alignment?

**Comment by rarkins** [comment]
Mockup showing a mix of both:


This issue contains a list of Renovate updates and their statuses.

## Awaiting Branch Creation

These branches can be created by Renovate on-demand once you click their checkbox below.

 - [ ] ☑️ build(deps): update dependency @aws-sdk/client-ecr to v3.16.0 
 - [ ] ☑️ build(deps): update dependency azure-devops-node-api to v11 
 - [ ] build(deps): update dependency crypto-random-string to v4 ☑️ 
 - [ ] build(deps): update dependency markdown-table to v3 ☑️ 
 - [ ] build(deps): update dependency p-map to v5 ☑️ 
 - [ ] build(deps): update dependency p-queue to v7 ☑️ 
 - [ ] build(deps): update dependency strip-ansi to v7 ☑️ 
 - [ ] 📆 chore(deps): update node.js to v16 ☑️ 
 - [ ] docs: update references to renovate/renovate 
 - [ ] docs: update references to renovate/renovate 📆 
 - [ ] chore(deps): lock file maintenance 📆 
 - [ ] 👶 chore(deps): update node.js to ed0af90 
 - [ ] 👶 build(deps): update dependency node-html-parser to v3.3.4 
 - [ ] chore(deps): update actions/cache action to v2.1.6 👶 
 - [ ] chore(deps): update dependency @types/eslint to v7.2.12 👶 
 - [ ] chore(deps): update dependency @types/node to v14.17.1 👶 
 - [ ] chore(deps): update jest monorepo (`@jest/globals`, `@jest/reporters`, `@jest/test-result`, `jest`, `jest-circus`) 👶 
 - [ ] build(deps): update dependency cacache to v15.2.0 👶 

## Branch Exists But Awaiting PR Creation

Tick any checkbox below to manually force creation of any of the following:

 - [ ] 🚦 build(deps): update dependency detect-indent to v6.1.0 
 - [ ] build(deps): update dependency shlex to v2.1.0 🚦 
 - [ ] chore(deps): update dependency jest-junit to v12.1.0 🚦 

## Open

 - [ ] 🚀 chore(deps): update dependency typescript to v4.3.2 
 - [ ] chore(deps): update linters (`@typescript-eslint/eslint-plugin`, `@typescript-eslint/parser`, `eslint-plugin-import`)

## Closed / Ignored

 - [ ] chore(deps): update node.js to v14.17.0

Reference:
☑️ Needs manual approval in dashboard
📆 Awaiting scheduled branch creation time
👶 Package is awaiting stabilityDays completion
🚦 Rate limited
🚀 Automerge enabled

**Comment by rarkins** [comment]
Reasons why I'm not sure to do that:

1. "tags" normally come after the information it's tagging, not before
2. We might have situations with more than one tag/emoji anyway

**Comment by viceice** [comment]
🤔 is it possible to use checkboxes in tables? so we use tables for alignment?

**Comment by rarkins** [comment]
It's not possible inside tables, last time I checked

**Comment by HonkingGoose** [comment]
> It's not possible inside tables, last time I checked

It's still not working (at least not on GitHub).

Example table with non-functional checkbox:

| Checkbox | Name | Emoji |
| --- | --- | --- |
| - [ ] | ESLint | ☑️ |

Raw Markdown:

```markdown
| Checkbox | Name | Emoji |
| --- | --- | --- |
| - [ ] | ESLint | ☑️ |
```

**Comment by HonkingGoose** [comment]
> Maybe move the emojis after checkbox for better alignment?

I like it better when the text is properly left-aligned. Also this way we can also add a Merge Confidence "confidence" badge at the end of the text as well:

- [ ] fix(deps): update docusaurus monorepo to v2.0.0-beta.0 🚦 [![confidence](https://badges.renovateapi.com/packages/npm/@docusaurus%2fcore/2.0.0-beta.0/confidence-slim/2.0.0-alpha.75)](https://docs.renovatebot.com/merge-confidence/)
- [ ] fix(deps): update eslint monorepo to v2.0.0 🚦 [![confidence](https://badges.renovateapi.com/packages/npm/@docusaurus%2fcore/2.0.0-beta.0/confidence-slim/2.0.0-alpha.75)](https://docs.renovatebot.com/merge-confidence/)
- [ ] fix(deps): update jest monorepo to v27 🚦 [![confidence](https://badges.renovateapi.com/packages/npm/@docusaurus%2fcore/2.0.0-beta.0/confidence-slim/2.0.0-alpha.75)](https://docs.renovatebot.com/merge-confidence/)

**Comment by viceice** [comment]
Ok, agreed. Was just an idea. 🤗

**Comment by HonkingGoose** [comment]
Stupid idea incoming, but would it be cleaner to removate the semantic commit prefixes, or even more text?

We can get pretty long lines as is, and those lines will only get longer if we eventually add Merge Confidence badges...

**rarkin's mockup:**

- [ ] build(deps): update dependency @aws-sdk/client-ecr to v3.16.0 ☑️
- [ ] build(deps): update dependency azure-devops-node-api to v11 ☑️
- [ ] build(deps): update dependency crypto-random-string to v4 ☑️
- [ ] build(deps): update dependency markdown-table to v3 ☑️
- [ ] build(deps): update dependency p-map to v5 ☑️
- [ ] build(deps): update dependency p-queue to v7 ☑️
- [ ] build(deps): update dependency strip-ansi to v7 ☑️
- [ ] chore(deps): update node.js to v16 ☑️
- [ ] docs: update references to renovate/renovate 📆
- [ ] docs: update references to renovate/renovate 📆
- [ ] chore(deps): lock file maintenance 📆
- [ ] chore(deps): update node.js to ed0af90 👶
- [ ] build(deps): update dependency node-html-parser to v3.3.4 👶
- [ ] chore(deps): update actions/cache action to v2.1.6 👶
- [ ] chore(deps): update dependency @types/eslint to v7.2.12 👶
- [ ] chore(deps): update dependency @types/node to v14.17.1 👶
- [ ] chore(deps): update jest monorepo (`@jest/globals`, `@jest/reporters`, `@jest/test-result`, `jest`, `jest-circus`) 👶
- [ ] build(deps): update dependency cacache to v15.2.0 👶

**Remove semantic commit prefixes**

- [ ] update dependency @aws-sdk/client-ecr to v3.16.0 ☑️
- [ ] update dependency azure-devops-node-api to v11 ☑️
- [ ] update dependency crypto-random-string to v4 ☑️
- [ ] update dependency markdown-table to v3 ☑️
- [ ] update dependency p-map to v5 ☑️
- [ ] update dependency p-queue to v7 ☑️
- [ ] update dependency strip-ansi to v7 ☑️
- [ ] update node.js to v16 ☑️
- [ ] update references to renovate/renovate 📆
- [ ] update references to renovate/renovate 📆
- [ ] lock file maintenance 📆
- [ ] update node.js to ed0af90 👶
- [ ] update dependency node-html-parser to v3.3.4 👶
- [ ] update actions/cache action to v2.1.6 👶
- [ ] update dependency @types/eslint to v7.2.12 👶
- [ ] update dependency @types/node to v14.17.1 👶
- [ ] update jest monorepo (`@jest/globals`, `@jest/reporters`, `@jest/test-result`, `jest`, `jest-circus`) 👶
- [ ] update dependency cacache to v15.2.0 👶

**Remove semantic commit prefixes + update text**

- [ ] dependency @aws-sdk/client-ecr to v3.16.0 ☑️
- [ ] dependency azure-devops-node-api to v11 ☑️
- [ ] dependency crypto-random-string to v4 ☑️
- [ ] dependency markdown-table to v3 ☑️
- [ ] dependency p-map to v5 ☑️
- [ ] dependency p-queue to v7 ☑️
- [ ] dependency strip-ansi to v7 ☑️
- [ ] node.js to v16 ☑️
- [ ] references to renovate/renovate 📆
- [ ] references to renovate/renovate 📆
- [ ] lock file maintenance 📆
- [ ] node.js to ed0af90 👶
- [ ] dependency node-html-parser to v3.3.4 👶
- [ ] actions/cache action to v2.1.6 👶
- [ ] dependency @types/eslint to v7.2.12 👶
- [ ] dependency @types/node to v14.17.1 👶
- [ ] jest monorepo (`@jest/globals`, `@jest/reporters`, `@jest/test-result`, `jest`, `jest-circus`) 👶
- [ ] dependency cacache to v15.2.0 👶

**Comment by rarkins** [comment]
"Remove semantic commit prefixes" is maybe a possibility. Maybe we can make it templated so that users can pick either.

**Comment by viceice** [comment]
Sounds good. I like the idea to have the text as short as possible.

**Comment by HonkingGoose** [comment]
@rarkins Do you want a separate issue for the "remove semantic commit prefixes" idea?

**Comment by rarkins** [comment]
I've hidden a bunch of comments here as I want to remain focused on the branch state caching first, and dashboard reuse of it second

**Comment by rarkins** [comment]
Blocked by #11426
